src/index.js

```diff
@@ -1,21 +1,33 @@
 "use strict";
 
-let WritableStream = require("./writable_stream");
+const WritableStream = require("./writable_stream");
 
-module.exports = class Renderer {
-	constructor(bundlePath) {
-		this.bundlePath = bundlePath;
-		this.reload();
-	}
+module.exports = rendererPath => {
+	return (req, res, next) => {
+		const renderer = typeof rendererPath === "function" ?
+			rendererPath :
+			resolveRenderer(res.app, rendererPath, { cache: true });
 
-	render(response, tag, params) {
-		// TODO: `response.status(200);`?
-		let stream = new WritableStream(response);
-		this._render(stream, tag, params);
-		response.end();
-	}
+		res.complate = (tag, params) => {
+			let stream = new WritableStream(res);
+			renderer(stream, tag, params);
+			res.end();
+		};
 
-	reload() {
-		this._render = require(this.bundlePath);
-	}
+		next();
+	};
 };
+
+function resolveRenderer(app, rendererPath, { cache } = {}) {
+	// by default the require cache is disabled for dev environment,
+	// so the given renderer is updated on each request.
+	// This enables a watch/compile setup for re-creating the renderer
+	// and its components.
+	//
+	// Set `cache` to true, if you want to cache the renderer also in dev-mode.
+	if(!cache && !app.enabled("view cache")) {
+		delete require.cache[rendererPath];
+	}
+
+	return require(rendererPath);
+}
```

Comment:
I've merged this part as 6b9621ec655a46b563621c92b33dc1776af27c5d (released as v0.9.0) because it moves things forward without getting bogged down by the remaining changes.

This commit also includes some adjustments WRT commit message, variable naming, conciseness (IMO anyway), plus it ensures `bundlePath` is absolute (I think? it's late and I'm tired... ).

Thank you for your patience; I'm somewhat overstretched at the moment.